src/libraries/System.Private.CoreLib/src/System/IO/Path.cs

```diff
@@ -306,17 +281,9 @@ public static bool HasExtension(string? path)
 
         public static bool HasExtension(ReadOnlySpan<char> path)
         {
-            for (int i = path.Length - 1; i >= 0; i--)
-            {
-                char ch = path[i];
-                if (ch == '.')
-                {
-                    return i != path.Length - 1;
-                }
-                if (PathInternal.IsDirectorySeparator(ch))
-                    break;
-            }
-            return false;
+            int i = PathInternal.GetExtensionStart(path);
+            // Found and not last char
+            return (i >= 0 && i < path.Length - 1);
```

Comment:
```suggestion
            return (i > 0 && i < path.Length - 1);
```
For `'.'` case, this makes it return on the first check rather than in the second.